### PR TITLE
[flutter_tools] fix tests that depend on correct cache existance

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

These tests will hit the real Cache, failing if the flutter root has been modified

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/hermetic/devices_test.dart [hunk 1]
```diff
@@ -23,9 +24,19 @@ void main() {
       Cache.disableLocking();
     });
 
+    MockCache cache;
+
+    setUp(() {
+      cache = MockCache();
+      when(cache.dyLdLibEntry).thenReturn(const MapEntry<String, String>('foo', 'bar'));
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/commands.shard/hermetic/devices_test.dart:31]: Device discovery created by instantiating DeviceDiscoverers looks up this entry from the cache, will fail if the path does not exist.

File: packages/flutter_tools/test/general.shard/device_test.dart [hunk 2]
```diff
@@ -18,6 +19,13 @@ import '../src/fake_devices.dart';
 import '../src/mocks.dart';
 
 void main() {
+  MockCache cache;
+
+  setUp(() {
+    cache = MockCache();
+    when(cache.dyLdLibEntry).thenReturn(const MapEntry<String, String>('foo', 'bar'));
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/device_test.dart:26]: Same here

File: packages/flutter_tools/test/general.shard/terminal_handler_test.dart [hunk 3]
```diff
@@ -27,6 +28,7 @@ void main() {
             trackWidgetCreation: false,
             treeShakeIcons: false,
           ),
+          generator: MockResidentCompiler(),
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/terminal_handler_test.dart:31]: If the generator is not provided, one is created but that looks up the dartSdk path from the cache.

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/terminal_handler_test.dart:31]: 🤦

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/terminal_handler_test.dart:31]: I thought it was hard to write tools tests, but turns out I had no idea...

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/terminal_handler_test.dart:31]: the nice thing is ... there are no issues with the tests using testWithoutContext, so at least we are on the right path 👍

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Pull request merged**